44.md

```diff
@@ -0,0 +1,96 @@
+NIP-44
+======
+
+Encrypted Direct Message (Versioned)
+------------------------------------
+
+`optional` `author:paulmillr`
+
+The NIP replaces NIP4, which is deficient with regards to algorithm choices, and introduces cryptography versioning. NIP4 is potentially vulnerable to [padding oracle attacks](https://en.wikipedia.org/wiki/Padding_oracle_attack) and uses keys which are not indistinguishable from random.
+
+A special event with kind `44`, meaning "encrypted direct message". It is supposed to have the following attributes:
+
+**`content`** MUST be equal to the CSV structure `v,param1,param2...`. Different versions may have different param count. First part is always algorithm version, a number. Params of version 1:
+
+  1. `nonce`: base64-encoded xchacha nonce
+  2. `ciphertext`: base64-encoded xchacha ciphertext, created from (key, nonce) against `plaintext`.
+     Example: `1,3dBKd83Pg2Q4Tu2A2e8N++c+ZW2IBc2f,FvQi1H4atMwU+FzUR/0CJ7kowjs+`
+
+**`tags`** MUST contain an entry identifying the receiver of the message (such that relays may naturally forward this event to them), in the form `["p", "<pubkey, as a hex string>"]`.
+
+**`tags`** MAY contain an entry identifying the previous message in a conversation or a message we are explicitly replying to (such that contextual, more organized conversations may happen), in the form `["e", "<event_id>"]`.
+
+**Note**: By default in the [libsecp256k1](https://github.com/bitcoin-core/secp256k1) ECDH implementation, the secret is the SHA256 hash of the shared point (both X and Y coordinates). We are using this exact implementation. In NIP4, unhashed shared point was used.
+
+## Metadata leakage
+
+All nostr events are passed over the internet, which means some metadata would always leak.
+
+All compatible clients MUST implement following flow to limit leakage:
+
+1. Alice authenticates on `wss://nostr.example.com` using NIP-42. This would mean sending and receiving
+   messages would only be available to proper, authenticated users.
+2. On success, Alice creates NIP-65 event specifying `wss://nostr.example.com` as her preferred relay.
```

Comment:
How about mentioning that these are Alice's `read` relays, the ones she reads from (in case she uses relays for just read or just write, instead of both).

... or better yet, since people are concerned about metadata, we extend NIP-65 to add a flag indicating a DM relay.  That way it would be possible for people concerned with metadata leakage to setup a personal relay for receiving DMs and be confident that those DMs are not being read by interlopers (at least the receipient is confident it doesn't leak from their specified read relay, but not that the sender didn't put it elsewhere; and the sender can't really be confident of this).